### ISSUE TITLE
Multiple email at once

### ISSUE THREAD

**Issue by andreaangeli** [rpremraj/mailR#58]
I really appreciate this great work but is there a solution if I'd like to send multiple emails to multiple recipients and I don't want the recipients see each other nor I want use the bcc field? I mean can I set a solution creating individual messages for each recipient from a draft message?

**Comment by rpremraj** [comment]
Yes of course. I do this regularly at my place of work.

Currently I don't see mailR performing this for you, but you can easily script to achieve what you need.

TIP: Look into the infuser package.

**Comment by matt-jay** [comment]
@andreaangeli: I implemented this as follows:

### Define distribution list
```
distrList <- list()

distrList[[]] <- list(
  recipients = c(...),
  <add any other parameters you want to distinguish between recipients>
)
<add as many of these as you want>
```

### Send e-mails to distribution list
```
for(i in 1:length(distrList)) {
    
    msgBody <- <message body>
    
    send.mail(
      from = "me@myself.com,
      to = c(unlist(list(distrList[[i]]$recipients))),
      subject = ...,
      body = msgBody,
      smtp = list(...),
      authenticate = TRUE,
      send = TRUE
   )    
  }
```

Hope this helps.

**Comment by andreaangeli** [comment]
@matt-jay I really appreciated your tips 👍  
but I'm not sure how to define my distribution list.
Is the following code correct?

`distrList <- list()
distrList[[]] <- list(
  recipients = c("email1@exemple.com", "email2@exemple.com"))`

When I try to run this code the console returns me:

`Error in distrList[[]] <- list(recipients = c("email1@exemple.com", "email2@exemple.com")) : 
  [[ ]] with missing subscript`

**Comment by matt-jay** [comment]
Apologies, tried to change something on hte fly there without checking. So you'll have to give each item an explicit index like so:

```
distrList <- list() 
distrList[['somename']] <- list( recipients = c("email1@exemple.com", "email2@exemple.com"))
```

Instead of a text label, you can also just use numbers. It feels to me like there is a more elegant way to doing this, but it does the job for now.

**Comment by andreaangeli** [comment]
@matt-jay awesome! Thanks a lot!
I've also improved you code as following:

`library(mailR)
setwd("C:/Users/User/Desktop/R/mail")

input <- "template.html"

distrList <- list()

distrList[['1']] <- list(recipients = c("EMAIL1@exemple.com"))
distrList[['2']] <- list(recipients = c("EMAIL2@exemple.com","EMAIL3@exemple.com"))


for(i in 1:length(distrList)) {
  

  
  send.mail(from = "MY_EMAIL@exemple.com",
            to = c(unlist(list(distrList[[i]]$recipients))),
            subject = "HTML SEND EXEMPLE",
            body = input,
            html = TRUE,
            smtp = list(host.name = "smtp.gmail.com", port = 465, user.name = "my_user", passwd = "my_pass", ssl = TRUE),
            authenticate = TRUE,
            send = TRUE)
}`

With this code I can send an html formatted email (the `input` function load an external html template that is located in the same directory) and I also added several `distrList` . So "EMAIL1@exemple.com" can't see I'm sending the same email, at the some time, also to "EMAIL2@exemple.com" and "EMAIL3@exemple.com" and vice versa. Instead "EMAIL2@exemple.com" can see I'm sending the same email at the same time to "EMAIL3@exemple.com" and vice versa
